apps/src/javalab/JavalabView.jsx

```diff
@@ -107,6 +138,59 @@ class JavalabView extends React.Component {
     });
   };
 
+  handleWidthResize = desiredWidth => {
+    let newWidth = Math.max(100, Math.min(desiredWidth, 600));
+    this.props.setLeftWidth(newWidth);
+
+    this.updateLayoutThrottled(newWidth);
+  };
+
+  updateLayout = width => {
+    const visualizationColumnHeight = $(window).height() - 135;
+    const visualizationTop = this.props.topInstructionsHeight;
+    const sliderHeight = 60;
+    let constrainVisualizationWidth =
+      visualizationColumnHeight - visualizationTop;
+    if (this.props.appType === 'neighborhood') {
+      constrainVisualizationWidth -= sliderHeight;
+    }
+    let newVizWidth = Math.min(constrainVisualizationWidth, width);
+
+    let scale = newVizWidth / 800;
+    if (scale < 0) {
+      // Avoid inverting.
+      scale = 0;
+    }
+    getStore().dispatch(setVisualizationScale(scale));
+
+    const cssScale = `scale(${scale})`;
+
+    if (this.props.appType === 'neighborhood') {
```

Comment:
We have renamed these apps a few times in the past. Where possible, keep using the enum to make future maintenance easier.